README.md

```diff
@@ -6,41 +6,184 @@
+Relation between these branches is determined by indentations - here single indent is 4 spaces but also tabulation(s) can be used.
+In the example above branches `allow-ownership-link` and `call-ws` are children of `develop` and `build-chain` is a child of `allow-ownership-link`.
+The same for `hotfix/add-trigger` and `master` - former is a child of latter.
+`develop` and `master` are root branches.
 
-The standard practice in Java logging is to use fully-qualified name of each class as category;
-in our case, however, we're only ever using the categories provided above.
-FQCN (and method name), however, is always a part of the log message itself.
+When branch tree is created, Git Machete indicates relation between parent branch and each of its child branches.
+If edge between them is green that means child branch is in sync with its parent branch - in other words there are no commits in parent branch that don't belong to the child.
+But if there are some of that commits, then edge is red - we need to [rebase](#rebase) child branch onto the parent.
 
-By default, IntelliJ logs everything with level `INFO` and above into `idea.log` file. <br/>
-The exact location depends on a specific IntelliJ installation; check `Help` -> `Show Log in Files` to find out. <br/>
-Tip: use `tail -f` to watch the log file as it grows.
+Machete file editor will help you to manage `machete` file: it underlines any errors (bad indentation or nonexistent branches) and propose branch names based on local repository branches.
+When file editing is done, you can click button in top right corner of file editor to refresh machete branch tree.
 
-To enable logging in `DEBUG` level, add selected categories to list in `Help` -> `Diagnostic Tools` -> `Debug Log Settings`. <br/>
-A relatively small amount of `TRACE`-level logs is generated as well.
+![](docs/machete_file_editor.gif)
 
+For more information about `machete` file look at the [reference blog post](https://medium.com/virtuslab/make-your-way-through-the-git-rebase-jungle-with-git-machete-e2ed4dbacd02)
 
-## Development
+### Checking out branches
+
+With Git Machete you can easily check out branches that have been defined in `machete` file.
+Just right-click on the branch you wont to check out and select `Checkout Branch`.
+You can also double-click on the selected branch to check it out.
+Current branch is underlined in branch tree.
+
+![](docs/checkout.gif)
+
+### Listing commits
+
+Git Machete can display commits belonging to the branches.
+It displays only these commits that are unique for the selected branch (i.e. were never a tip of a different branch).
+When you click on eye button on the left side of plugin, you can toggle between show and hide commits.
+
+![](docs/toggle_listing_commits.gif)
+
+
+### Rebase
+
+Rebase with this plugin is easy!
```

Comment:
`this plugin` does not sound good to me :confused: I fell kind of distance from OUR (:heart:) plugin. Why don't we reference `Git Machete` by its name?